### ISSUE TITLE
Injection on ViewController looks broken with Xcode 7.1 beta

### ISSUE THREAD

**Issue by benjamincombes** [Swinject/Swinject#10]
Hi,
I had an application working with Swinject, but since I tried to compile it with Xcode 7.1, it stopped working, because all of the ViewControllers are not injected with values.
The ViewControllers is correctly created, but all values that should be injected are left to nil ; did something change, or did I do something wrong?
Let me know if you need any details, thanks a lot, here are some code samples (extracts only) :

```swift
container.register(AssetModel.self) {
            r in return AssetModel()
        }.inObjectScope(.Hierarchy)

container.register(DemoModel.self) {
            r in DemoModel()
        }.inObjectScope(.Hierarchy)

container.registerForStoryboard(PartnersViewController.self) {
            r, c in
            c.assetModel = r.resolve(AssetModel.self)
            c.demoModel = r.resolve(DemoModel.self)
            c.playViewController = self.slotMachineStoryboard.instantiateViewControllerWithIdentifier(ViewControllerIdentifiers.Play.rawValue) as? PlayViewController
        }

container.registerForStoryboard(PlayViewController.self) {
            r, c in
            c.assetModel = r.resolve(AssetModel.self)
            c.demoModel = r.resolve(DemoModel.self)
        }

let slotMachineStoryboard = SwinjectStoryboard.create(name: StoryboardIdentifiers.SlotMachine.rawValue, bundle: nil, container: mainContainer)
let initialViewController = slotMachineStoryboard.instantiateViewControllerWithIdentifier(ViewControllerIdentifiers.Partners.rawValue) as! PartnersViewController
```

**Comment by yoichitgy** [comment]
I also experienced the problem. Some unit tests of Swinject related to Storyboard started failing since Xcode 7.1. It looks Apple has changed something about Storyboard.

Still I'm investigating the problem.

**Comment by yoichitgy** [comment]
Actually the cast `entry.initCompleted as? (Resolvable, C) -> ()` in `Container.swift` fails with Xcode 7.1. Still investigating.

```swift
internal func runInitCompleted<C: Controller>(controllerType: C.Type, controller: C, name: String? = nil) {
    resolutionPool.incrementDepth()
    defer { resolutionPool.decrementDepth() }
    
    let key = ServiceKey(factoryType: controllerType, name: name)
    if let entry = getEntry(key) {
        resolutionPool[key] = controller as Any
        if let completed = entry.initCompleted as? (Resolvable, C) -> () {
            completed(self, controller)
        }
    }
}
```

**Comment by yoichitgy** [comment]
Fixed. Still no tagged version is released, but you can install the fixed version specifying the commit ed97226 or later in `xcode7.1` branch.

**Comment by benjamincombes** [comment]
Great, thanks a lot, I'll give it a try!
